.github/workflows/container-test.yaml

```diff
@@ -0,0 +1,48 @@
+---
+name: Container Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and Push
```

Comment:
Should this step be skipped when testing or somehow put into test only mode?

Reply:
It will only push tags like this

```yaml
      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v5
        with:
          images: |
            ghcr.io/${{ github.repository }}
          tags: |
            type=ref,event=branch
            type=ref,event=pr
```

It will have the branch and pr number as a tag

Reply:
I need this spelled out for me a little more. My impression is there are two actions here;
1) The other one, which pushes the container to the registry when a new release is made
2) This one which is meant to test for a PR that changes the container. It will be built but not pushed. 

In (2) what makes it not attempt to push to the container registry? I see it trying to upload.  Are you saying that is working as intended and its pushing development tags/ branches for each PR attempt?

Reply:
> In (2) what makes it not attempt to push to the container registry? I see it trying to upload. Are you saying that is working as intended and its pushing development tags/ branches for each PR attempt?

Yes, the second one will only push images on push to main and when you open a pull request. So the result images will look like this:

```sh
ghcr.io/allenporter/flux-local:pr-1315 # Anytime you open a PR
ghcr.io/allenporter/flux-local:main    # Anytime you merge to main
ghcr.io/allenporter/flux-local:v1.     # Anytime you create a release
ghcr.io/allenporter/flux-local:v1.0    # Anytime you create a release
ghcr.io/allenporter/flux-local:v1.0.0  # Anytime you create a release
```

I also made the "Container Test" workflow only run when you open a PR since outside contributors do not have access to your secrets, otherwise the workflow would fail. I hope that clears it up.

Reply:
OK got it. I've never seen (2) before, but i can see why that is useful.